pages/price-feeds/use-real-time-data/ton.mdx

```diff
@@ -124,6 +124,112 @@ This code snippet does the following:
+
+They both are demonstrated in the [Pyth Connector example](https://github.com/pyth-network/pyth-examples/tree/main/price_feeds/ton/pyth-connector). <br/>
+These same patterns are also used in the  [EVAA Protocol code](https://github.com/evaafi/contracts/tree/v8) for implementing following operations:
+- Pyth proxy pattern: liquidate TON / supply_withdraw TON.
+- Onchain-getter pattern: liquidate jetton / supply_withdraw jetton.
+
+Choose the pattern that best fits your use case and how you want to handle assets and price updates in your application.
+
+Each operation described above can result in either a successful outcome or an error. It is important to consider and handle both scenarios for every pattern.
+
+## Pyth proxy: Success
+In the EVAA protocol, the operations that implement the Pyth proxy pattern are <b>`liquidate (TON)`</b> and <b>`supply_withdraw (TON)`</b>. In these cases, the user sends a request to the Pyth contract using the native TON asset. As a result of the operation, the user may receive either TON or JETTON tokens back, depending on the outcome of the transaction.
+
+```mermaid
+sequenceDiagram
+  autonumber
+  participant "U" as "User"
+  participant "P" as "Pyth Contract"
+  participant "M" as "EVAA Master"
+
+  note over "M": master.fc:121 — received from Pyth (op 5)
+  "U"->>"P": request (0x3 liquidate_master or 0x4 supply_withdraw_master)
+  "P"-->>"M": op 5 parse_price_feed_updates (prices)
+  note over "M": Master accepts and processes the transaction
+```
+
+- Related code (GitHub):
+  - [master.fc: entry for Pyth message](https://github.com/evaafi/contracts/blob/d9138cb24f03b53522774351aceb38c51a047eee/contracts/master.fc#L121)
+  - [Pyth proxy: supply_withdraw (TON) in master.fc](https://github.com/evaafi/contracts/blob/d9138cb24f03b53522774351aceb38c51a047eee/contracts/master.fc#L192-L211)
+  - [Pyth proxy: liquidate (TON) in master.fc](https://github.com/evaafi/contracts/blob/d9138cb24f03b53522774351aceb38c51a047eee/contracts/master.fc#L171-L190)
+
+
+## Pyth proxy: Error handling
+In the Pyth proxy pattern, when an error occurs (i.e., Pyth cannot process the request and sends a `response_error` with op 0x10002), the error report is sent directly back to the user who initiated the transaction, not to a contract. This is different from the on-chain getter pattern, where the error is returned to the EVAA Master contract for further handling and potential refund logic. In the proxy case, the user receives the error response from the Pyth contract, including the error code and the original query ID, allowing the user to identify and handle the failure on their side.
+
+```mermaid
+sequenceDiagram
+  autonumber
+  participant "U" as "User"
+  participant "P" as "Pyth Contract"
+
```

Comment:
The double quotes can be removed from here. Else these are coming in the diagram.

<img width="749" height="279" alt="image" src="https://github.com/user-attachments/assets/f651cb27-3aa0-4fdc-a365-f4e4960234e5" />

Reply:
done